client.go

```diff
@@ -0,0 +1,168 @@
+// Package kafkatest provides a package intended for running tests
+// that require a Kafka backend.
+package kafkatest
+
+import (
+	"crypto/rand"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Shopify/sarama"
+)
+
+var ErrDisabled = fmt.Errorf("kafka tests are disabled")
+
+// New connects to a Kafka instance and returns a Kafka
+// instance that uses it.
+//
+// The following environment variables can be used to
+// configure the connection parameters:
```

Comment:
Two main questions I have for this way of working:

* Why are you using environment variables instead of arguments to the builder? 
* Why aren't you using confita for this?

The main I suppose I have is the lack of context of knowing how this tool would work :innocent: so replying these questions could help me to understand it better.

Reply:
> Why are you using environment variables instead of arguments to the builder?

It's intended to be a significant part of the value added by this package. Because this is designed to be a way of easily doing kafka-based tests - we don't want to have to encode the environment-variable-getting logic in every test suite.

> Why aren't you using confita for this?

IMHO confita is overkill for this. Confita is useful to allow a flexible backend for fetching config values, but in this case we only care about environment variables. I don't think this code would be any simpler using confita. Additionally, I don't really want to pull in all the external dependencies that confita uses.

> The main I suppose I have is the lack of context of knowing how this tool would work :innocent: so replying these questions could help me to understand it better.

The clue is in the package name :) This is intended to be used in a test suite to enable easy connection to a kafka cluster in a standard way, and automatic creation and destruction of kafka topics in a way that doesn't interfere with other tests running in parallel.

Reply:
Thanks a lot for the explanations! it's clearer now. I think we can leave the package as it is now  let's see how it evolves